sqlglot/optimizer/merge_subqueries.py

```diff
@@ -174,6 +174,26 @@ def _outer_select_joins_on_inner_select_join():
             for col in inner_projections[selection].find_all(exp.Column)
         )
 
+    def _is_recursive():
+        # Recursive CTEs look like this:
+        #     WITH RECURSIVE cte AS (
+        #       SELECT * FROM ...
+        #       UNION ALL
+        #       SELECT *
+        #       FROM cte
+        #     )
+        # In this case, the inner scope is SELECT * FROM ..., NOT the entire CTE.
+        # So we traverse up to the inner scope's parent.
```

Comment:
@barakalon what do you mean here? not sure I get it

Reply:
will update comment